flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/delegation/ParserImpl.java

```diff
@@ -101,6 +113,29 @@ public ResolvedExpression parseSqlExpression(String sqlExpression, TableSchema i
                 sqlExpressionExpanded);
     }
 
+    public String[] getCompletionHints(String statement, int cursor) {
+        List<String> candidates =
+                new ArrayList<>(
+                        Arrays.asList(
+                                ParseStrategyParser.INSTANCE.getCompletionHints(
```

Comment:
Could you add a final member variable `extendedParser` instead of referencing `ParseStrategyParser.INSTANCE` so many times?